bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
  exit 1
 end.parse!
 
-class State
- #Loads the config file
- def initialize(state)
-  @statefile = state 
-  @sf = YAML.load_file(@statefile)
- end
+#If no command was entered display help and exit 
+if options.empty?
+ puts opt_parser
+ exit 1
+end
 
- #Toggles value in state hash
- def toggle(name)
-  #Finds all true/false values in hash
-  t = @sf[name].key(true)
-  f = @sf[name].key(false) 
+module ConfigFile
+ class Config
+  #Load the config file
+  def initialize
+   configfile = File.join($path, 'config.yml')
+   @config = YAML.load_file(configfile)
+  end
+  
+  #Check if a setting has been set
+  def setting_exists?(setting)
+   if @config[setting] != nil then
+    return true 
+   else
+    puts "The config file does not contain a path for the setting: #{setting}"
+    exit 1 
+   end 
+  end
+  
+  #Set the directory for the job   
+  def jobdir(name)
+   setting_exists?('jobdir')
+   @jobdir = @config['jobdir']
+   
+   #Check if the path given in the file is absolute or relative   
```

Comment:
Determining if the path give in the config is absolute or relative, is relevant to all the config parameters, not just jobdir. I'd suggest looking into doing a private method to expand to a full path and call that with the config parameter. eg:
```
def jobdir(name)
  jobdir=validate_path(@config[:jobdir])
  ...
end

private

def validate_path
  #confirm the full path here
end
```

Reply:
Moved the path checking to a private method that can take any given path, so all current config settings can now make use of it